plugins/exercism/api_key.go

```diff
@@ -0,0 +1,94 @@
+package exercism
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://exercism.org/docs/using/solving-exercises/working-locally"),
+		ManagementURL: sdk.URL("https://exercism.org/settings/api_cli"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.URL,
+				MarkdownDescription: `The URL of the Exercism API.`,
+				Secret:              false,
+			},
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Exercism.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 37,
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                sdk.FieldName("Directory"),
+				MarkdownDescription: `The path to the workspace directory where the exercises are stored.`,
+				Secret:              false,
+			},
+		},
+		DefaultProvisioner: provision.TempFile(
+			tempFileConfig,
+			provision.AtFixedPath("~/.config/exercism/user.json"),
```

Comment:
For many tools, the equivalent of `~/.config` from Linux on MacOS is `~/Library/Application Support`.

Is this the case here too?

Reply:
I actually had to find the config file on my Mac since the documentation doesn't give the location, so it is indeed `~/.config` for MacOS.